fragment.go

```diff
@@ -1389,6 +1399,94 @@ func (f *fragment) bulkImport(rowIDs, columnIDs []uint64) error {
 	return unprotectedWriteToFragment(f, results)
 }
 
+// bulkImportMutex performs a bulk import on a fragment while ensuring
+// mutex restrictions. Because the mutex requirements must be checked
+// against storage, this method must acquire a write lock on the fragment
+// during the entire process, and it handles every bit independently.
+func (f *fragment) bulkImportMutex(rowIDs, columnIDs []uint64) error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	// Disconnect op writer so we don't append updates.
+	f.storage.OpWriter = nil
+
+	// Process every bit.
+	// If an error occurs then reopen the storage.
+	lastID := uint64(0)
+	if err := func() error {
+		rowSet := make(map[uint64]struct{})
+		for i := range rowIDs {
+			rowID, columnID := rowIDs[i], columnIDs[i]
+
+			// Handle mutex vector (i.e. clear an existing row).
+			if existingRowID, found := f.mutexVector.Get(columnID); found && existingRowID != rowID {
+				// Determine the position of the bit in the storage.
+				pos, err := f.pos(existingRowID, columnID)
+				if err != nil {
+					return err
+				}
+
+				// Clear storage.
+				_, err = f.storage.Remove(pos)
+				if err != nil {
+					return err
+				}
+
+				rowSet[existingRowID] = struct{}{}
+			}
+
+			// Determine the position of the bit in the storage.
+			pos, err := f.pos(rowID, columnID)
+			if err != nil {
+				return err
+			}
+
+			// Write to storage.
+			_, err = f.storage.Add(pos)
+			if err != nil {
+				return err
+			}
+
+			// Reduce the StatsD rate for high volume stats
+			f.stats.Count("ImportBit", 1, 0.0001)
+
+			// import optimization to avoid linear foreach calls
+			// slight risk of concurrent cache counter being off but
+			// no real danger
```

Comment:
can we expunge this comment from the face of the earth? does anyone truly know its arcane meaning?

Reply:
how many PR's you gonna comment on this comment?

Reply:
I think this comment should be something like:
```
`rowSet` maintains the set of rowIDs present in this import.
It allows the cache to be updated once per row, instead of once
per bit. In doing this, there is a slight risk of the cache counter
being off due to the fact that there may be multiple updates
to the row data before the cache is updated.
```

Reply:
With that said, is there a possibility of something concurrently accessing the cache during an import? It seems like the write lock on the fragment would prevent that.